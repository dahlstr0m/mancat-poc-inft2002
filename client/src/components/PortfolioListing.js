// @flow

import * as React from 'react';
import { Component } from 'react-simplified';
import { projectService, posterService } from '../services';
import { PortfolioCard } from './Card';
import { Alert } from './Widgets';

/**
 * Renders portfolio list.
 */
export default class PortfolioListing extends Component {
  projects: Project[] = [];
  posters: Poster[] = [];

  render() {
    return (
      <>
        {this.projects
          .filter((project) => project.active === true) // Filter to ensure only Active projects will be displayed
          .sort((projectA, projectB) => projectA.ranking - projectB.ranking) // Sorting to display by asc ranking
          .map((
            project // Mapping to display all left projects with PortfolioCard widget & fetching correct corresponding thumbnail for url attrib.
          ) => (
            <PortfolioCard
              key={project.projectId}
              projectId={project.projectId}
              title={project.title}
              link={'/projects/' + project.projectId}
              imageUrl={
                this.posters.find((poster) => poster.projectId === project.projectId)
                  ? this.posters.find((poster) => poster.projectId === project.projectId)
                      .thumbnailUrl
                  : ''
              }
            >
              {'ingen data'}
            </PortfolioCard>
          ))}
      </>
    );
  }

  mounted() {
    projectService
      .getProjects()
      .then((projects) => (this.projects = projects))
      .catch((error: Error) => Alert.danger('Error getting projects: ' + error.message));

    posterService
      .getPosters()
      .then((posters) => (this.posters = posters))
      .catch((error: Error) => Alert.danger('Error getting posters: ' + error.message));
  }
}
