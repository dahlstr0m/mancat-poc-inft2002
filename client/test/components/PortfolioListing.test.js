// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

// Components
import PortfolioListing from '../../src/components/PortfolioListing';

// Mocks
jest.mock('../../src/services/project-service');
jest.mock('../../src/services/poster-service');
jest.mock('../../src/services/category-service');

// Tests
describe('PortfolioListing component tests', () => {
  test('PortfolioListing draws correctly', (done) => {
    const wrapper = shallow(<PortfolioListing />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();

      done();
    });
  });
});
