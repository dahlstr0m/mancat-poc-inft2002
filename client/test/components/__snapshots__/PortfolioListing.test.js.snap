// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioListing component tests PortfolioListing draws correctly 1`] = `
<Fragment>
  <Card>
    <FormLabel>
      Search:
    </FormLabel>
    <FormInput
      onChange={[Function]}
      type="text"
      value=""
    />
    <FormLabel>
      Category:
    </FormLabel>
    <FormSelect
      onChange={[Function]}
      value={0}
    >
      <option
        key="0"
        value={0}
      >
        All
      </option>
      <option
        key="1"
        value={1}
      >
        Kategori 1
      </option>
      <option
        key="2"
        value={2}
      >
        Kategori 2
      </option>
      <option
        key="3"
        value={3}
      >
        Kategori 3
      </option>
      <option
        key="4"
        value={4}
      >
        Kategori 4
      </option>
    </FormSelect>
  </Card>
  <CardGrid
    columns={1}
    columnsLg={4}
    columnsMd={3}
    columnsSm={2}
  >
    <CardColumn
      key="1"
    >
      <CardImageEffect
        img="Poster 1 thumb url"
        imgAlt="Missing thumbnail for Project 1"
        imgHeight={300}
        imgWidth={210}
        link="/projects/1"
      />
    </CardColumn>
    <CardColumn
      key="3"
    >
      <CardImageEffect
        img="Poster 3 thumb url"
        imgAlt="Missing thumbnail for Project 3"
        imgHeight={300}
        imgWidth={210}
        link="/projects/3"
      />
    </CardColumn>
  </CardGrid>
</Fragment>
`;
